selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/intelligent/CommandsEditor.java

```diff
@@ -324,85 +324,152 @@ public void waitTabIsNotPresent(String nameOfFile) {
+   * Waits until command tab with specified {@code commandName} is displayed with the unsaved
+   * status.
    *
-   * @param nameCommand name of tab command
+   * @param commandName title of the commands editor's tab
    */
-  public void waitTabCommandWithUnsavedStatus(String nameCommand) {
+  public void waitTabCommandWithUnsavedStatus(String commandName) {
     seleniumWebDriverHelper.waitVisibility(
-        By.xpath(format(ACTIVE_TAB_UNSAVED_FILE_NAME, nameCommand)), ELEMENT_TIMEOUT_SEC);
+        By.xpath(format(ACTIVE_TAB_UNSAVED_FILE_NAME, commandName)), ELEMENT_TIMEOUT_SEC);
   }
 
-  public void typeTextIntoNameCommandField(String nameCommand) {
-    seleniumWebDriverHelper.setValue(nameCommandField, nameCommand);
+  /**
+   * Types specified {@code commandName} to the "Command Name" field, and waits until typed text is
+   * displayed.
+   *
+   * <p>Note! Method clears previous value in the field.
+   *
+   * @param commandName the command name which should be typed
+   */
+  public void typeTextIntoNameCommandField(String commandName) {
+    seleniumWebDriverHelper.setValue(nameCommandField, commandName);
   }
 
+  /**
+   * Waits until specified {@code expectedText} is displayed in the "Command Name" field.
+   *
+   * @param expectedText text which should be displayed
+   */
   public void waitTextIntoNameCommandField(String expectedText) {
     seleniumWebDriverHelper.waitValue(
         nameCommandField, expectedText, REDRAW_UI_ELEMENTS_TIMEOUT_SEC);
   }
 
+  /**
+   * Waits until specified {@code expectedText} is displayed in the "Goal" field.
+   *
+   * @param expectedText text which should be displayed
+   */
   public void waitTextIntoGoalField(String expectedText) {
     seleniumWebDriverHelper.waitText(nameGoalField, expectedText, REDRAW_UI_ELEMENTS_TIMEOUT_SEC);
   }
 
+  /**
+   * Opens "Goal" drop-down list and clicks on specified {@code goalName}.
+   *
+   * @param goalName item's visible name in the "Goal" drop-down
+   */
   public void selectGoalNameIntoCommandEditor(String goalName) {
     seleniumWebDriverHelper.waitAndClick(iconDropDownGoal, REDRAW_UI_ELEMENTS_TIMEOUT_SEC);
     seleniumWebDriverHelper.waitVisibility(goalDropDown, REDRAW_UI_ELEMENTS_TIMEOUT_SEC);
     seleniumWebDriverHelper.waitAndClick(By.xpath(format(NAME_IN_GOAL_DROP_DOWN, goalName)));
   }
 
+  /**
+   * Clicks on specified {commandsEditorType} in the "Commands Editor".
+   *
+   * @param commandsEditorType command editor's type which defined in {@link CommandsEditorType}
+   */
   public void setFocusIntoTypeCommandsEditor(String commandsEditorType) {
     seleniumWebDriverHelper.waitAndClick(
         By.xpath(commandsEditorType), REDRAW_UI_ELEMENTS_TIMEOUT_SEC);
     editor.waitActive();
   }
 
+  /**
+   * Waits until specified {@code expectedText} is displayed in the popup with macro description.
+   *
+   * <p>Note! Popup with description opens after single click on item in the macro autocomplete
```

Comment:
looks misleading - like method itself click on item, but it doesn't.
I would remove it so as there could be another way to open macro form.